### PR TITLE
ART-3050: add plashet from-tags --rhcos

### PR DESCRIPTION
**Pull request by vfreex**
This PR adds `--rhcos` option to Plashet from-tags subverb.
When "--rhcos" argument is specified, the final list of package NVRs should include any dependencies specified in the assembly's assembly.rhcos.dependencies field.

### CODE REVIEW

**Comment by vfreex** [comment]
@jupierce This PR is ready to review. I added a new function `assembly_rhcos_config` in `assembly.py` to handle `rhcos` field inheritance. Let me know if this is the desired behavior. Thank you in advance.

File: doozerlib/plashet.py [hunk 1]
```diff
@@ -170,11 +170,39 @@ def from_image_member_deps(self, el_version: int, assembly: str, releases_config
...
+                raise IOError(f"The following image member dependency NVRs don't exist: {missing_nvrs}")
             # Make sure image member dependencies have no ART managed rpms.
             art_rpms_in_deps = {dep_build["name"] for dep_build in dep_builds} & {meta.rpm_name for meta in rpm_map.values()}
             if art_rpms_in_deps:
                 raise ValueError(f"Unable to build plashet. Image member dependencies cannot have ART managed RPMs: {art_rpms_in_deps}")
             for dep_build in dep_builds:
                 component_builds[dep_build["name"]] = dep_build
         return component_builds
+
+    def from_rhcos_deps(self, el_version: int, assembly: str, releases_config: Model, rpm_map: Dict[str, RPMMetadata]):
```
Comment (jupierce) [inline doozerlib/plashet.py:182]:
```suggestion
    def from_rhcos_deps(self, el_version: int, assembly: str, releases_config: Model, rpm_map: Dict[str, Dict]):
```
Just brew build dicts, right?

File: doozerlib/plashet.py [hunk 2]
```diff
@@ -170,11 +170,39 @@ def from_image_member_deps(self, el_version: int, assembly: str, releases_config
...
+        """ Returns RPM builds defined in RHCOS config dependencies
+        :param el_version: RHEL version
+        :param assembly: Assembly name to query. If None, this method will return true latest builds.
+        :param releases_config: a Model for releases.yaml
+         :param rpm_map: Map of rpm_distgit_key -> RPMMetadata
```
Comment (jupierce) [inline doozerlib/plashet.py:187]:
```suggestion
        :param rpm_map: Map of rpm_distgit_key -> RPMMetadata
```

**Review by jupierce — CHANGES_REQUESTED** [review]
Just nits. Otherwise lgtm.

**Comment by jupierce** [comment]
Going to go ahead and merge with tweaks. I would like to use `assembly_rhcos_config` in my next story.

**Pull request merged**